modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/model/listener/test/CompanyModelListenerTest.java

```diff
@@ -0,0 +1,252 @@
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.UnicodePropertiesBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class CompanyModelListenerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		Company company = CompanyTestUtil.addCompany();
+
+		_companyId = company.getCompanyId();
+
+		_user = UserTestUtil.getAdminUser(_companyId);
+	}
+
+	@Test
+	public void testOnBeforeDelete() throws Exception {
+		long originalCompanyId = CompanyThreadLocal.getCompanyId();
+
+		CompanyThreadLocal.setCompanyId(_companyId);
+
+		NotificationContext notificationContext = _createNotificationContext();
+
+		NotificationTemplate notificationTemplate =
+			_notificationTemplateLocalService.addNotificationTemplate(
+				notificationContext);
+
+		Assert.assertEquals(_companyId, notificationTemplate.getCompanyId());
+
+		NotificationRecipient notificationRecipient =
+			notificationTemplate.getNotificationRecipient();
+
+		Assert.assertEquals(_companyId, notificationRecipient.getCompanyId());
+
+		List<NotificationRecipientSetting> notificationRecipientSettings =
+			notificationRecipient.getNotificationRecipientSettings();
+
+		for (NotificationRecipientSetting notificationRecipientSetting :
+				notificationRecipientSettings) {
+
+			Assert.assertEquals(
+				_companyId, notificationRecipientSetting.getCompanyId());
+		}
+
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				"A" + RandomTestUtil.randomString(),
+				Collections.singletonList(
+					new TextObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"a" + RandomTestUtil.randomString()
+					).build()),
+				ObjectDefinitionConstants.SCOPE_COMPANY, _user.getUserId());
+
+		_objectActionLocalService.addObjectAction(
+			RandomTestUtil.randomString(), _user.getUserId(),
+			objectDefinition.getObjectDefinitionId(), true, StringPool.BLANK,
+			RandomTestUtil.randomString(),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			RandomTestUtil.randomString(),
+			ObjectActionExecutorConstants.KEY_NOTIFICATION,
+			ObjectActionTriggerConstants.KEY_ON_AFTER_ADD,
+			UnicodePropertiesBuilder.put(
+				"notificationTemplateId",
+				notificationTemplate.getNotificationTemplateId()
+			).build(),
+			false);
```

Comment:
Do we really need to create an object action so we can add a NotificationQueueEntry? Did you check how NotificationQueueEntryLocalServiceTest is creating?

Reply:
I didn't. I only check the UserNotificationTypeTest and EmailNotificationTypeTest
I will check this one you mentioned, thanks